pkg/network/prober/prober.go

```diff
@@ -52,36 +53,42 @@ func WithHost(host string) Preparer {
 
 // ExpectsBody validates that the body of the probe response matches the provided string.
 func ExpectsBody(body string) Verifier {
-	return func(r *http.Response, b []byte) (bool, error) {
-		return string(b) == body, nil
+	return func(r *http.Response, b []byte) error {
+		if string(b) == body {
+			return nil
+		}
+		return fmt.Errorf("unexpected body: want %v got %v", body, string(b))
```

Comment:
```suggestion
		return fmt.Errorf("unexpected body: want %q got %q", body, string(b))
```

Reply:
nit: Add a `,`: `want %q, got %q"`
Same above and below.